runelite-client/src/main/java/net/runelite/client/plugins/nightmarezone/NightmareZoneOverlay.java

```diff
@@ -0,0 +1,156 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.nightmarezone;
+
+import net.runelite.api.Client;
+import net.runelite.api.ItemID;
+import net.runelite.api.Varbits;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.components.BackgroundComponent;
+import net.runelite.client.ui.overlay.components.TextComponent;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.awt.*;
+
+class NightmareZoneOverlay extends Overlay
+{
+	private static final int WIDTH = 140;
+	private static final int TOP_BORDER = 2;
+	private static final int BOTTOM_BORDER = 3;
+
+	private final Client client;
+	private final NightmareZoneConfig config;
+	private final NightmareZonePlugin plugin;
+	private final InfoBoxManager infoBoxManager;
+	private final ItemManager itemManager;
+
+	private AbsorptionCounter absorptionCounter;
+
+	@Inject
+	NightmareZoneOverlay(
+			@Nullable Client client,
+			NightmareZoneConfig config,
+			NightmareZonePlugin plugin,
+			InfoBoxManager infoBoxManager,
+			ItemManager itemManager
+	)
+	{
+		setPosition(OverlayPosition.TOP_LEFT);
+		setPriority(OverlayPriority.LOW);
+		this.client = client;
+		this.config = config;
+		this.plugin = plugin;
+		this.infoBoxManager = infoBoxManager;
+		this.itemManager = itemManager;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		if (!config.enabled() || !plugin.isInNightmareZone() || !config.moveOverlay())
+		{
+			if (absorptionCounter != null)
+			{
+				removeAbsorptionCounter();
+				// Restore original widget height
+				Widget nmzWidget = client.getWidget(WidgetInfo.NIGHTMARE_ZONE);
+				if (nmzWidget != null)
+				{
+					nmzWidget.setHidden(false);
+				}
+			}
+			return null;
+		}
+
+		Widget nmzWidget = client.getWidget(WidgetInfo.NIGHTMARE_ZONE);
+
+		if (nmzWidget != null)
+		{
+			nmzWidget.setHidden(true);
+		}
+
+		String scoreString = "Points: " + client.getSetting(Varbits.NMZ_POINTS);
+
+		renderAbsorptionCounter();
+
+		FontMetrics fm = graphics.getFontMetrics();
+
+		int height = TOP_BORDER + fm.getHeight() + BOTTOM_BORDER; // opponent name
+
+		final BackgroundComponent backgroundComponent = new BackgroundComponent();
```

Comment:
Instead of using backgound component together with text component here, you can just use PanelComponent